c_glib/arrow-dataset-glib/dataset.cpp

```diff
@@ -152,12 +153,43 @@ gadataset_dataset_to_table(GADatasetDataset *dataset, GError **error)
   }
   auto arrow_scanner = *arrow_scanner_result;
   auto arrow_table_result = arrow_scanner->ToTable();
-  if (!garrow::check(error, arrow_scanner_result, "[dataset][to-table]")) {
+  if (!garrow::check(error, arrow_table_result, "[dataset][to-table]")) {
```

Comment:
This is an unrelated change but I noticed this was checking the wrong result, so if `arrow_scanner_builder->Finish()` succeeded but `arrow_scanner->ToTable()` failed the error would not have been correctly caught.